src/Data/Variant/Internal.purs

```diff
@@ -109,14 +114,30 @@ lookup
   → L.List String
   → L.List a
   → a
-lookup name tag = go
+lookup name tag tags as = fromJust name $ lookup' tag tags as
+
+fromJust
+  ∷ ∀ a
+  . String
+  → M.Maybe a
+  → a
+fromJust name = case _ of
+  M.Just a → a
+  _ → unsafeCrashWith $ "Data.Variant: impossible `" <> name <> "`"
+
+lookup'
+  ∷ ∀ a
+  . String
+  → L.List String
+  → L.List a
+  → M.Maybe a
+lookup' tag = go
```

Comment:
I don't think these changes are necessary given other feedback. I'd like to avoid the `Just` box since we just immediately remove it.